File: Regression/Random Forest Regression/random_forest_regression.py
# -*- coding: utf-8 -*-
"""
Created on Thu Jan 19 11:55:23 2017

@author: rakes
"""

# Random Forest Regression
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values

# Random Forest Model
from sklearn.ensemble import RandomForestRegressor
regressor = RandomForestRegressor(n_estimators=100, random_state=0, 
                                  max_features= 'sqrt')
regressor.fit(X, y)

X_grid = np.arange(min(X), max(X), 0.01)
X_grid = X_grid.reshape((len(X_grid), 1))
plt.scatter(X, y, color = 'red')
plt.plot(X_grid, regressor.predict(X_grid))
plt.title('Salary Vs Label (Random Forest)')
plt.xlabel('Label')
plt.ylabel('Salary')
plt.show()

# Predict salary of a new employee
value = regressor.predict(np.array([[6.5]]))
print(value)

File: Regression/Random Forest Regression/random_forest_regression.R
# Random Forest - Regression

# Import the Data set
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

# Fitting the Random Forest Regression model to the dataset
library(randomForest)
set.seed(1234)
regressor = randomForest(formula = Salary ~ ., data = dataset, ntree = 100)

# Predicting the salary of a new employee
prediction = predict(regressor, data.frame(Level = 6.5))

# Plotting
library(ggplot2)
X_grid = seq(min(dataset$Level), max(dataset$Level), 0.01)
ggplot() + geom_point(aes(dataset$Level, dataset$Salary)) +
  geom_line(aes(X_grid, predict(regressor, data.frame(Level = X_grid)), color = 'red')) +
  ggtitle('Salary Vs Level (Random Forest)') + xlab('Level') + ylab('Salary')